scripts/test/test-cli.sh

```diff
@@ -30,8 +30,18 @@ ${PORTER_HOME}/porter invoke --action=plan --debug
 
 ${PORTER_HOME}/porter upgrade --debug --param file_contents='bar!'
 
-echo "Verifying installation output(s) via 'porter installation output show' after upgrade"
-${PORTER_HOME}/porter installation output show file_contents | grep -q "bar!"
+echo "Verifying installation output via 'porter installation output show' after upgrade"
+# Verify the output matches the expected value
+if [[ "$(${PORTER_HOME}/porter installation output show file_contents)" != "bar!" ]]; then
+  echo "Output value: '${output}' does not match expected"
+  exit 1
+fi
+
+# Verify the output has no extra newline (mixin should trim newline added by terraform cli)
```

Comment:
Is the extra check here necessary now that we are doing the exact equality check above? If there was a newline, wouldn't the `!= "bar"` catch it?

Reply:
Great question.  Alas, bash's default command substitution strips trailing newline characters.  I've been looking into ways to work around this (https://stackoverflow.com/questions/15184358/how-to-avoid-bash-command-substitution-to-remove-the-newline-character has been particularly helpful), but they seem to quickly grow ungainly.  I also see that I should create a function to validate a given output, as we do so after install as well; so I'll at least update to DRY that up...